Source/Flow/Private/FlowAsset.cpp

```diff
@@ -189,7 +189,22 @@ void UFlowAsset::HarvestNodeConnections()
 		}
 	}
 }
-#endif
+
+void UFlowAsset::AddCustomInput(const FName& InName)
```

Comment:
_Accessors for subclasses to use to modify the CustomInput array_

Reply:
2 minor questions here
- Is it considered an editor-only code? I'd wrap it with `WITH_EDITOR`
- Would you see any problem with changing `checks` in Add/Remove method to simply ensure or even if? I tend to avoid potentially crashing checks if isn't necessary. Here nothing changes if a "wrong" event name would be provided.

Reply:
1. AddCustomInput() could be Editor-only, if it's only ever expected to be done in editor.
2. I tend to use a lot of asserts (checks), but you could downgrade them to ensureRuntimeWarning or something.  ensure() is just a check() that is never stripped from the code, so it will still appear to crash.   Since you have your own messagelog in Flowgraph, you might write an error to there instead.

I'll be more careful to avoid check() in the future for this codebase.

Do you want me to include those changes in my next submission or will you make them?

Reply:
I'll do it.

However... what if I'd promoted the `CustomInputs` and `CustomOutputs` properties to protected? We wouldn't have to add these methods at all in the plugin. And it might be easier to maintain for you?

Reply:
Somebody on the discord made a good point about keeping them private and providing accessors. So I think it’s reasonable to keep the accessor pattern.

Reply:
Changes committed! Thanks again! :)